### ISSUE TITLE
No usage docs

### ISSUE THREAD

**Issue by dov** [nyorain/vkpp#3]
Vkpp comes without any usage docs, tests, or samples. Reading the headers is not sufficiently clear. As an example consider the ```VKPP_CALL``` macro, which requires understanding quite a bit of ```call.hpp``` just to figure out what are its three arguments ```(d, f, ...)```.

**Comment by nyorain** [comment]
Good point. I didn't really intend the macro to be used manually but the parameters should be documented (or better named) anyways, will fix it. Otherwise it's hard to say which parts need documentation though since this just automatically generates headers...
Do you have any other code that should be better documented? And any ideas for tests that could be useful here? I already included the basic "the headers compile" test.

**Comment by dov** [comment]
I think the documentation should be aimed at someone who wants to **use** vkpp for writing a Vulkan program. What is the philosophy of translating the the vulkan api into C++, what macros are used for exceptions, etc. Perhaps some more practical comparison of the differences vs Vulkan-hpp .

A nice example would be to following the idea from https://github.com/vsg-dev/vsgExamples/tree/master/Desktop/vsgdraw and port https://github.com/Overv/VulkanTutorial/blob/master/code/26_depth_buffering.cpp to vkpp.

A more minimal example that is just doing a hello triangle with glfw would be nice as well.

And a compute example. :smile:

**Comment by nyorain** [comment]
This was initially just a small script-like generator i hacked together for vpp because I didn't like vkpp. But yeah, you are totally right, such examples would be useful (and I could show off some compilation time comparison with Vulkan-hpp in the README, the main thing that disturbs me about Vulkan-hpp). I added some first docs now but will hopefully find the time to come back to this!
